### DISCUSSION TITLE
New plugin for generating gcode from vector graphics

### DISCUSSION THREAD

**Discussion by breiler** [winder/Universal-G-Code-Sender#1634] [Show and tell]
The new plugin for generating gcode using simple vector graphics (#1409) has been merged and is available for testing in the latest nightly build (https://github.com/winder/Universal-G-Code-Sender#downloads). 

Here is a small demo of the feature:
https://youtu.be/iUi6Lr8nZCo

Give it a try and let me know of the biggest pains or missing features!

**Comment by MilanNemsak** [comment]
Hello, I am using UGS(Windows) - GRBL(Arduino).  I wanted to import DXF file. It works for lines and circles with one problem - the conversion does not consider units. UGS has units [mm], DXF file has $INSUNITS = 4 but data are represented in .ugsd like [inch]. Did I something wrong? 
You write here about Arcs. Is it possible import Arc from DXF file? I did not achieve it. Thanks for answer.
                                                                      Milan

**Reply by breiler** [reply]
Hi @MilanNemsak,

The softwares that I used for generating DXF:s did not include units or the `$INSUNITS` parameter. As of UGS 2.0.13 it should import the file in the same units as UGS is using. So if running UGS with MM it will try to import the file as MM.

Can you share a sample that I can test with both for arcs and the units?

**Comment by MilanNemsak** [comment]
Thanks for realy fast answer! I try be similar.  Milan

**Comment by MilanNemsak** [comment]
Hi Joacim,

I attache test file, cca 40-th line is $INSUNITS, next numbers "70; 4" .
Number "4" is for [mm],  "1"  is for inches.
DXF file is produced by 2D  QCAD from RIBBONSOFT.
Here data about UGS

*Product Version:* Universal Gcode Platform 20220810

*Java:* 13.0.1; OpenJDK 64-Bit Server VM 13.0.1+9

*Runtime:* OpenJDK Runtime Environment 13.0.1+9

*System:* Windows 7 version 6.1 running on amd64; Cp1252; en_US
(ugsplatform)

*User directory:* C:\Users\Milan\.ugsplatform\2.0.12\dev

*Cache directory:* C:\Users\Milan\.ugsplatform\2.0.12\dev\var\cache


I am not very experienced, I am not able send you a copy of the screen
after import. I try to do it tomorrow.

The arcs are omitted and lines are 25.4 times longer.

**Reply by breiler** [reply]
The 2.0.12 version of UGS is a bit old, plenty of changes has been made to the designer since then. However I haven't made any changes to the DXF-importer for Arcs so that implementation is probably missing, so I'll be happy to have a look at your DXF-sample.

Unfortunatley you can't attach files when replying through emails, you need to do that from the github-page.

**Reply by MilanNemsak** [reply]
Hi Joacim,
promised attached
[PrintScreen.odt](https://github.com/winder/Universal-G-Code-Sender/files/10857662/PrintScreen.odt)

**Comment by MilanNemsak** [comment]
I hope it passed but it was necessary change .dxf to .txt
[JBreierTest1txt.txt](https://github.com/winder/Universal-G-Code-Sender/files/10853043/JBreierTest1txt.txt)
[JBreierTest1.pdf](https://github.com/winder/Universal-G-Code-Sender/files/10853044/JBreierTest1.pdf)

**Reply by breiler** [reply]
Thanks, I am currently busy with another project. I'll maybe have time and look at this tomorrow.

**Comment by MilanNemsak** [comment]
I am retired. I have enough time :-) Don't worry.  Thanks for a lesson of English. Milan

**Comment by MilanNemsak** [comment]
Hi Joacim,

big sorry for the distortion of your surname.  I did few tests as you recommended:
- download version 2.0.13 and 2.0.14
- import DXF and SVG
- 2.0.13 and 2.0.14 DXF result : lines are in correct size, correct position, arcs omitted , 
- 2.0.13 and  2.0.14 SVG result: lines and arcs present in correct shape, not correct size and position - it is possible to change it to any size and position by UGS ( well done ).
 What am I doing? I try to save an old small milling machine. It was on the road to the scrap.  It is only a toy and for my joy.

                                                                               Milan

**Reply by breiler** [reply]
Sorry for the delayed answer. I have added support for parsing DXF Arcs and now remember why I skipped it in the first place. It was more difficult than I would like to admit... 😭 

There are still some issues, read about them here: https://github.com/winder/Universal-G-Code-Sender/pull/2171

It is now available in the latest nightly build of UGS found here:
https://github.com/winder/Universal-G-Code-Sender#downloads

**Comment by MilanNemsak** [comment]
Thanks, I try it soon. Milan
